multiview/embed/mvmds.py

```diff
@@ -0,0 +1,239 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from multiview.embed.base import BaseEmbed
+from multiview.utils.utils import check_Xs
+import warnings
+import numpy as np
+from sklearn.metrics import euclidean_distances
+
+
+class MVMDS(BaseEmbed):
+
+    """"
+    An implementation of Classical Multiview Multidimensional Scaling for
+    jointly reducing the dimensions of multiple views of data. A Euclidean
+    distance matrix is created for each view, double centered, and the k l
+    argest common eigenvectors between the matrices are returned based on
+    the stepwise estimation of common principal components.
+
+    Parameters
+    ----------
+    n_components : int (positive), default=None
+        Represents the number of components that the user would like to
+        be returned from the algorithm. This value must be greater than
+        0 and less than the number of samples within each view.
+
+    num_iter: int (positive), default=15
+        Number of iterations stepwise estimation goes through. Detailed
+        in Trendafilov paper.
+
+    References
+    ----------
+    .. [#1] Trendafilov, Nickolay T. “Stepwise Estimation of Common Principal
+            Components.” Computational Statistics &amp; Data Analysis, vol. 54,
+            no. 12, 2010, pp. 3446–3457., doi:10.1016/j.csda.2010.03.010.
+
+    """
+
+    def __init__(self, n_components=None, num_iter=15):
+
+        super().__init__()
+        self.components = None
+        self.n_components = n_components
+        self.num_iter = num_iter
+
+    def _commonpcs(self, Xs):
+
+        """
+        Finds Stepwise Estimation of Common Principal Components as described
+        by common Trendafilov implementations based on the following paper:
+
+        https://www.sciencedirect.com/science/article/pii/S016794731000112X
+
+        Parameters
+        ----------
+
+        Xs: List of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+
+        Returns
+        -------
+
+        components: The first n_component primary principal components
```

Comment:
Format the return correctly. Aka first line should be the return type and below it is the description.

Reply:
Specifying the shape is also important here

Reply:
Didn't see this being addressed?